src/engine.cc

```diff
@@ -166,8 +168,20 @@ SearchLimits EngineController::PopulateSearchLimits(int ply, bool is_black,
   return limits;
 }
 
-void EngineController::UpdateNetwork() {
+void EngineController::UpdateTBAndNetwork() {
   SharedLock lock(busy_mutex_);
+
+  std::string tb_paths = options_.Get<std::string>(kSyzygyTablebaseStr);
+  if (!tb_paths.empty() && tb_paths != tb_paths_) {
+    tb_paths_ = tb_paths;
+    syzygy_tb_ = std::make_unique<SyzygyTablebase>();
+    std::cerr << "Loading Syzygy tablebases from " << tb_paths_ << std::endl;
+    if (!syzygy_tb_->init(tb_paths_)) {
```

Comment:
Maybe throw Exception in this case? I think it's better to refuse doing anything if path is provided but fails to load, rather than silently ignore.

`tb_paths_` is better to update as a last thing, so it's still null if it fails.
Also it's better to write code more exception-friendly:

```cpp
auto tb = std::make_unique<SyzygyTablebase>(tb_path);
if (!tb->init) throw Exception(blah blah)
syzygy_tb_ = std::move(tb);
tb_paths_ = tb_paths;
```

Reply:
I just tested Stockfish, it behaves exactly like I wrote it, just prints how many TB files it found, and if it's zero, it's zero. (Most importantly, neither this code nor Stockfish is "silently ignoring" it.)

There's a lot of other config besides just the TB, and it would be a real pain to users if they had to restart the program and fill in all other settings just because they typo'd the syzygy path.

Reply:
It's much more painful to run engine thinking that TBs are there while they are not.